addon/search/searchcursor.js

```diff
@@ -115,13 +115,61 @@
     }
   }
 
+  //Normalization for Arabic
+  var normalizeArabicChars = function (s) {
+    function filter(c) {
+      switch (c) {
+        // ALEF Chars
+        case 'إ' :
+        case 'أ' :
+        case 'آ' :
+        case 'ٵ' :
+        case 'ٳ' :
+        case 'ٲ' :
+        case 'ٱ' :
+          return 'ا'
+        // TAAA MARBOTA Chars
+        case 'ۃ' :
+        case 'ہ' :
+          return 'ة'
+        // YAAA Chars
+        case 'ى' :
+        case 'ي' :
+        case 'ٸ' :
+          return 'ي'
+        case 'ئ':
+          return 'ي ء'
```

Comment:
Is this space in the returned string intentional?

Reply:
Also, `.normalize("NFD") already separates this character into \u064a and \u0654, which seems similar to what you're doing, and might already cover this?

Reply:
Thanks for comment .... yes, you are right, it seems that when I converted each Unicode into its char I added a space by mistake .... sorry for that.
About normalize("NFD"), Actually I tested it but it didn`t work , but for sure if you test it, I can remove this case to reduce checks.